lib/super_good/solidus_taxjar/transaction_id_generator.rb

```diff
@@ -0,0 +1,45 @@
+module SuperGood
+  module SolidusTaxjar
+    # Responsible for generating `transaction_id` references for transactions
+    # we create on TaxJar for Solidus orders. This class handles creating
+    # associated ID's for transactions which need to be cancelled and recreated
+    # when the order is updated in Solidus after it has been sent to the TaxJar
+    # reporting API.
+    class TransactionIdGenerator
+      class << self
+        # Generates the next sequential `transaction_id` given an order and
+        # optionally the current transaction ID on TaxJar. This handles the
+        # case where a transaction already has been created on TaxJar and later
+        # needs to be cancelled and we need to create an updated transaction
+        # with an associated identifier.
+        #
+        # @param order [Spree::Order] the order for which we want to generate a
+        #   transaction ID.
+        # @param current_transaction_id [String] the current transaction ID for
+        #   the order if it exists on TaxJar.
+        # @return [String] the next sequential `transaction_id`
+        def next_transaction_id(order:, current_transaction_id: nil)
+          if current_transaction_id.nil?
+            "#{order.number}"
+          elsif order.number == current_transaction_id
+            "#{current_transaction_id}-1"
+          else
+            parts = current_transaction_id.split(/-/)
```

Comment:
You could use [`rpartition`](https://ruby-doc.org/core-2.5.1/String.html#method-i-rpartition) here instead which would remove the need for the `[0..-2].join("-")` below.